src/excel_loader.py

```diff
@@ -0,0 +1,236 @@
+import tkinter as tk
+import pandas as pd
+import os
+import openpyxl
+import json
+from tkinter import filedialog, ttk, simpledialog
+from openpyxl import Workbook
+from openpyxl.utils.dataframe import dataframe_to_rows
+from openpyxl.styles import NamedStyle
+from datetime import time
+from models import TaskType
+
+input_file_paths = []
+input_schedule_days = []
+
+
+def create_window():
+    root = tk.Tk()
+    root.title("Study Schedule Selection")
+    root.geometry("500x400")  # Set the initial size of the window (width x height)
+
+    file_select_button = tk.Button(root, text="Select Schedule", command=open_file_dialog)
+    file_select_button.pack(pady=10)
+
+    global file_table
+    columns = ("Study Schedule", "Day")
+    file_table = ttk.Treeview(root, columns=columns, show="headings", selectmode="browse")
+
+    # Set column headings
+    for col in columns:
+        file_table.heading(col, text=col)
+        file_table.column(col, width=150, anchor="center")
+
+    file_table.pack(pady=5)
+
+    generate_schedule_button = tk.Button(root, text="Generate Schedules", command=output_directory_dialog)
+    generate_schedule_button.pack(pady=10)
+
+    root.mainloop()
+
+
+def open_file_dialog():
+    file_path = filedialog.askopenfilename(filetypes=[("Excel files", "*.xlsx *.xls")])
+    input_file_paths.append(file_path)
+
+    file_name = file_path.split("/")[-1]  # Get only the filename from the path
+
+    # Prompt the user to enter the day number using a dialog box
+    day = simpledialog.askinteger("Enter Day", f"Enter the day for '{file_name}':")
+    input_schedule_days.append(day)
+
+    # Add the file name and the entered number to the table
+    file_table.insert("", "end", values=(file_name, day))
+
+
+def output_directory_dialog():
+    output_file_path = filedialog.askdirectory()
+    copy_schedule_days(input_file_paths, output_file_path)
+
+
+def find_token_row_index(token, df):
+    return df[df.apply(lambda row: row.astype(str).str.contains(token, case=False).any(), axis=1)].index
+
+
+def append_empty_rows(sheet, num_rows):
+    for _ in range(num_rows):
+        empty_row = [None] * sheet.max_column
+        sheet.append(empty_row)
+
+
+def copy_schedule_days(input_files, output_file_directory):
```

Comment:
This function does quite a bit, would pay to refactor - maybe separate this out into a different python module (.py file) and break it into smaller functions

Reply:
Ah I see, the `input_schedule_days` array is going to make it a bit tricky.

To get past that you'd need to pass that in as a parameter to the function (and not as a module level variable / global). That can be a bit tricky with the current layout with the tk UI. Can help you refactor in person, or do a writeup here if you want to do it at work (lmk)

Reply:
no lets do in person so I can explain it properly :p